airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py

```diff
@@ -1298,13 +1298,13 @@ class PropertyHistory(Stream):
     """
 
     more_key = "has-more"
-    url = "/contacts/v1/lists/recently_updated/contacts/recent"
+    url = "/contacts/v1/lists/all/contacts/all"
```

Comment:
that depends on how much a lot is :)

Unfortunately it's impossible because of the API signature. It does not allow us to filter out records by date(time) field. It only supports pagination by vid-Offset (unique contact Id).
Besides, the API returns contacts, containing all the history of their properties changes, each of which we retrieve as a single record:
```
{
  "vid": 3234574,
  "canonical-vid": 3234574,
  "merged-vids": [3234574,407546],
  "portal-id": 62515,
  "is-contact": true,
  "profile-token": "AO_T-GsyaBk",
  "profile-url": "https://app.hubspot.com/contacts/62515/contact/12630274",
  "properties": {
    "createdate": {
      "value": "1484026585538",
      "versions": [
        {
          "value": "1484026585538",
          "source-type": "API",
          "source-id": "Contacts",
          "source-label": null,
          "timestamp": 1484026585538,
          "selected": false
        },
        {
          "value": "1484026586699",
          "source-type": "API",
          "source-id": "Contacts",
          "source-label": null,
          "timestamp": 1484026586699,
          "selected": false
        }
      ]
    }
    ...
``` 
-->>
```
        {
          "value": "1484026585538",
          "source-type": "API",
          "source-id": "Contacts",
          "source-label": null,
          "timestamp": 1484026585538,
          "selected": false,
          "property": "createdate",
          "vid": 3234574
        },
        {
          "value": "1484026586699",
          "source-type": "API",
          "source-id": "Contacts",
          "source-label": null,
          "timestamp": 1484026586699,
          "selected": false,
          "property": "createdate",
          "vid": 3234574
        }
```
The same vid, different cursor values, no filters supported by the API.
The best we can do here is a "semiincremental" mode which would anyway read all the data and then filter out old records.

Reply:
p.s. if you worry about all the data being read within one request - that's not the case, it is paginated